src/docker/build.rs

```diff
@@ -0,0 +1,125 @@
+use std::env;
+use std::process::Command;
+use std::str::FromStr;
+
+use super::engine::Engine;
+use crate::errors::*;
+use crate::shell::Verbosity;
+
+#[derive(Debug, PartialEq, Eq, Clone, Copy)]
+enum Progress {
+    Plain,
+    Auto,
+    Tty,
+}
+
+impl FromStr for Progress {
+    type Err = eyre::ErrReport;
+
+    fn from_str(progress: &str) -> Result<Self> {
+        Ok(match progress {
+            "plain" => Progress::Plain,
+            "auto" => Progress::Auto,
+            "tty" => Progress::Tty,
+            s => eyre::bail!("unexpect progress type: expected plain, auto, or tty and got {s}"),
+        })
+    }
+}
+
+impl From<Progress> for &str {
+    fn from(progress: Progress) -> Self {
+        match progress {
+            Progress::Plain => "plain",
+            Progress::Auto => "auto",
+            Progress::Tty => "tty",
+        }
+    }
+}
+
+pub trait DockerCommandExt {
+    fn build(&mut self) -> &mut Self;
+    fn progress(&mut self, progress: Option<&str>) -> Result<&mut Self>;
+    fn verbose(&mut self, verbosity: Verbosity) -> &mut Self;
+    fn scan_suggest(&mut self, suggest: bool) -> &mut Self;
+    fn cross_labels(&mut self, target: &str, platform: &str) -> &mut Self;
+}
+
+impl DockerCommandExt for Command {
+    fn build(&mut self) -> &mut Self {
+        match Engine::has_buildkit() {
+            true => self.args(["buildx", "build"]),
+            false => self.arg("build"),
+        }
+    }
+
+    fn progress(&mut self, progress: Option<&str>) -> Result<&mut Self> {
+        let progress: Progress = match progress {
+            None => env::var("CROSS_BUILD_PROGRESS")
```

Comment:
I don't like parsing from `Progress` and then re-serializing it, but this also needs good error messages with clap. This needs to be changed to use the parser from clap with auto-derive parsers and serializers.

Reply:
I think changing it take `Option<Progress>` otherwise parsing the environment variable is the best option, and letting clap use the default parser. Otherwise, we'd need to use `clap` or `serde` logic internally.